sqlmesh/lsp/reference.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+
+from lsprotocol.types import Range, Position
+import typing as t
+
+from sqlmesh.core.dialect import normalize_model_name
+from sqlmesh.core.model.definition import SqlModel
+from sqlmesh.lsp.context import LSPContext
+from sqlglot import exp
+
+from sqlmesh.utils.pydantic import PydanticModel
+
+
+class Reference(PydanticModel):
+    range: Range
+    uri: str
+
+def get_model_definitions_for_a_path(lint_context: LSPContext, document_uri: str) -> t.List[Reference]:
+    """
+    Get the model references for a given path.
+
+    Works for models and audits.
+    Works for targeting sql and python models.
+
+    Steps:
+    - Get the parsed query
+    - Find all table objects using find_all exp.Table
+        - Match the string against all model names
+    - Need to normalize it before matching
+    - Try get_model before normalization
+    - Match to models that the model refers to
+    """
+    # Ensure the path is a sql model
+    if not document_uri.endswith(".sql"):
+        return []
+    
+    # Get the model 
+    models = lint_context.map[document_uri]
+    if models is None: 
+        return []
+    if len(models) == 0:
+        return []
+    model_name = models[0]
+    model = lint_context.context.get_model(model_or_snapshot=model_name, raise_if_missing=False)
+    if model is None: 
+        return []
+    if not isinstance(model, SqlModel):
+        return []
+    
+    # Find all possible references
+    tables = list(model.query.find_all(exp.Table))
+    if len(tables) == 0:
+        return []
+    
+    references = []
+    for table in tables:
+        depends_on = model.depends_on
+
+        # Normalize the table reference
+        reference_name = table.this.this if table.db is None else f"{table.db}.{table.this.this}"
```

Comment:
I think this doesn't work if the model name was quoted explicitly by the user. I think you want to do `table.sql(dialect=model.dialect)` instead.